enhancements/kubelet/user-namespaces-support.md

```diff
@@ -0,0 +1,454 @@
+
+##### nested-container
+
+The intention of this SCC is to allow a user to run `podman` or other container engine inside of an Openshift pod.
+Since user namespaces allow a process to gain access to the capabilities needed in a safe way, it's a natural addition
+to the proposal of adding user namespaces generally.
+
+This SCC will largely mirror the `restricted-v2` SCC, but have a couple of changes.
+- SELinux context set to MustRunAs.SeLinuxOptions.Type: `container_engine_t`
+    - This SELinux type has been developed to allow the majority of podman in pod situations, and can
+      continue to be adapted without affecting the normal `container_t` which should be more restrictive.
+- RunAsUserStrategy: RunAsAny
+    - Any user should be allowed, as the user running in the container is not the same running outside.
+- RequiredDropCapabilities: None
+    - Inside of a user namespace, the capabilities a pod requests are only present in the user namespace,
+      not on the host. Thus, even for a less trusted user, the capabilities should be safe to access.
+- `UserNamespaceLevel` set to `RequirePodLevel`
+
+Note: to use this SCC, the namespace must be labeled as `privileged` in Pod Security Standards. This is in part because
+PSS doesn't recognized `container_engine_t` [in 1.30](https://github.com/kubernetes/kubernetes/pull/126165), and in part because
+the baseline policy doesn't allow Unmasked procMount, even if the pod is in a user namespace, [in 1.30](https://github.com/kubernetes/kubernetes/pull/126163).
+
+These are fixed in Kubernetes 1.31, and Openshift 4.18, meaning a namespace can be labeled as `baseline` and successfully utilize all the fields of the `nested-container` SCC.
+
+#### Feature Gates and Sets
+
+There are three feature gates of note for this enhancement: `UserNamespacesSupport`, `UserNamespacesPodSecurityStandards`, and `ProcMountType`.
+All three of these feature gates are upstream Kubernetes features.
+
+- `UserNamespacesSupport` is the main toggle for allowing user namespaces in Kubernetes. Without it, the kubelet and kube-apiserver will filter the `hostUsers` field from a pod spec.
+- `UserNamespacesPodSecurityStandards` is for toggling the relaxation of Pod Security Standards when a pod is in a user namespace. This feature will remain in alpha, until the oldest supported kubelet
+  of the newest kube-apiserver supports denying a pod when the feature is enabled but the kernel doesn't support user namespaces.
+- `ProcMountType` is a feature for allowing an `Unmasked` `/proc` mount in a container. While not directly related to user namespaces on the surface, the feature relies on the presence of a user namespace.
+  When paired with a user namespace, the feature opens the opportunity for a container to mount `/proc` mounts internally, which is useful for running nested containers.
+
+`UserNamespacesSupport` will be on by default OOTB in 4.17, as it's beta in 1.30. It is the main featuregate of this feature.
```

Comment:
Looks to be disabled by default: in 1.30 and 1.31

1. 1.30 https://github.com/kubernetes/kubernetes/blob/release-1.30/pkg/features/kube_features.go#L1231
2. 1.31 https://github.com/kubernetes/kubernetes/blob/release-1.31/pkg/features/kube_features.go#L1220

Reply:
updated